llvm/lib/SYCLLowerIR/ComputeModuleRuntimeInfo.cpp

```diff
@@ -152,7 +152,8 @@ std::optional<T> getKernelSingleEltMetadata(const Function &Func,
 
 PropSetRegTy computeModuleProperties(const Module &M,
                                      const EntryPointSet &EntryPoints,
-                                     const GlobalBinImageProps &GlobProps) {
+                                     const GlobalBinImageProps &GlobProps,
+                                     module_split::IRSplitMode SplitMode) {
```

Comment:
i would prefer we didnt add another argument to this function, can we instead use module metadata like we do for for sycl/esimd split (`saveSplitInformationAsMetadata`)?